process/block/metablock.go

```diff
@@ -174,22 +168,31 @@ func (mp *metaProcessor) ProcessBlock(
 		mp.headersCounter.getNumShardMBHeadersTotalProcessed(),
 	)
 
-	requestedShardHdrs, requestedFinalShardHdrs := mp.requestShardHeaders(header)
+	mp.createBlockStarted()
+	requestedShardHdrs, requestedFinalityAttestingShardHdrs := mp.requestShardHeaders(header)
 
 	if haveTime() < 0 {
 		return process.ErrTimeIsOut
 	}
 
-	if requestedShardHdrs > 0 || requestedFinalShardHdrs > 0 {
-		log.Info(fmt.Sprintf("requested %d missing shard headers and %d final shard headers\n", requestedShardHdrs, requestedFinalShardHdrs))
+	haveMissingShardHeaders := requestedShardHdrs > 0 || requestedFinalityAttestingShardHdrs > 0
+	if haveMissingShardHeaders {
+		log.Info(fmt.Sprintf("requested %d missing shard headers and %d finality attesting shard headers\n",
+			requestedShardHdrs,
+			requestedFinalityAttestingShardHdrs))
+
 		err = mp.waitForBlockHeaders(haveTime())
-		mp.mutRequestedShardHdrsHashes.Lock()
-		mp.allNeededShardHdrsFound = true
-		unreceivedShardHdrs := len(mp.requestedShardHdrsHashes)
-		mp.mutRequestedShardHdrsHashes.Unlock()
+
+		mp.hdrsForCurrBlock.mutHdrsForBlock.Lock()
```

Comment:
make a common function in baseProcess and call that from shardProcessor and metaProcessor as well.

Reply:
Done